plugins/modules/luks_device.py

```diff
@@ -904,6 +1009,22 @@ def run_module():
     if module.params['label'] is not None and module.params['type'] == 'luks1':
         module.fail_json(msg='You cannot combine type luks1 with the label option.')
 
+    if module.params['remove_keyslot'] is not None:
+        if module.params['keyfile'] is None and module.params['passphrase'] is None:
+            module.fail_json(msg='Removing a keyslot requires the passphrase or keyfile of another slot')
+
+    luks_type = crypt.get_luks_type(conditions.get_device_name())
```

Comment:
This is now queried every time that the module is called, even if this information is never needed. It would be better to only query it when needed.